src/Components/ProductFilters.jsx

```diff
@@ -1,12 +1,39 @@
-import { Disclosure, Menu, Transition } from "@headlessui/react";
-import { ChevronDownIcon, FilterIcon } from "@heroicons/react/solid";
-import React, { Fragment } from "react";
+import { Disclosure, Menu, Transition } from '@headlessui/react';
+import { ChevronDownIcon, FilterIcon } from '@heroicons/react/solid';
+import React, { Fragment, useEffect } from 'react';
+import { useState } from 'react';
 
 function classNames(...classes) {
-  return classes.filter(Boolean).join(" ");
+  return classes.filter(Boolean).join(' ');
 }
 
-export default function ProductFilters({ filterOptions, setFilterOptions, sortOptions, setSortOptions }) {
+export default function ProductFilters({
+  filterOptions,
+  setFilterOptions,
+  sortOptions,
+  setSortOptions,
+  products,
+  setProducts,
+  allproducts,
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**nitpick:** ‏allproducts => allProducts